### DISCUSSION TITLE
Certain X-SAMPA characters (`{`) don't work

### DISCUSSION THREAD

**Discussion by hmturnbull** [espeak-ng/espeak-ng#1809] [Q&A] [answered]
Hi.  I've been playing around with espeak-ng, and it seems to be just what I was looking for.  But I've been having a problem where certain X-SAMPA characters don't work; the speech synthesizer just stops suddenly.  Notably, the [æ] phoneme, written as `{`, isn't recognized, despite espeak-ng's documentation saying that's the character for that vowel.  Same with others, like `&` and quite a few modifiers such as `_r`.

If I enter a command like `espeak-ng -l fonxsamp "[['pain{:pl=]]"` (for this example, I used the word "pineapple" , then it gets to the end of "pine" and stops.

- I'm using PowerShell 7.3.6 on Windows 10 (64-bit).

It's possible I'm missing something, since I'm still _relatively_ new to both espeak-ng and X-SAMPA, so any help would be much appreciated.

**Comment by jaacoppi** [comment, marked as answer]
It's a bug in the implementation. ReadClause() and TranslateClause() are famously buggy and in need of refactoring.

If you want to try to fix it yourself, check how ReadClause() cuts sentences in smaller pieces and see why it stops processing.

**Reply by hmturnbull** [reply]
Thanks very much.  Good to know it's a known bug.  I don't have C experience, unfortunately.  If this is a known bug, should I assume there's already an issue for it, or should I submit that separately?

**Comment by jaacoppi** [comment]
There are quite a few issues about the ReadClause() parser, but not necessarily about SAMPA specifically. I think we can leave this open as a reminder.
